test/test_torch.py

```diff
@@ -16313,6 +16319,40 @@ def tracker(worker):
 ---(input size: {:4}, eigenpairs:{:2}, units: relative error, maxiter={:4})---
 '''.format(tol, eq_err, eq_err_general, iters1, eq_err_scipy, eq_err_general_scipy, iters2, m, k, niter))
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @precisionOverride({torch.bfloat16: 1e-1})
+    @dtypes(*(torch.testing.get_all_dtypes()))
+    def test_outer(self, device, dtype):
```

Comment:
This is a good test. Let's add a test for some type promotion behavior two. This can be done in a couple ways:

- you can hand write a few cases (e.g. long x float32)
- you can take a product in dtypes `@dtypes(*product(torch.testing.get_all_dtypes(), torch.testing.get_all_dtypes()))` and determine the appropriate result type using `torch.result_type` on the inputs you construct, then verify the function produces the correct inputs

Reply:
Also: this test, and any other tests for addr and outer, should go in test_linalg.py.

Reply:
Thanks for the detailed suggestions, the test for type promotion now takes a product of `get_all_dtypes` and `get_all_dtypes`, and manipulate `torch.result_type` function to determine the appropriate result type.